pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -195,7 +195,12 @@ func initIdentity(set *apps.StatefulSet, pod *v1.Pod) {
 func updateIdentity(set *apps.StatefulSet, pod *v1.Pod) {
 	pod.Name = getPodName(set, getOrdinal(pod))
 	pod.Namespace = set.Namespace
-
+	if pod.Spec.Hostname == "" {
+		pod.Spec.Hostname = pod.Name
```

Comment:
Why isn't this reading the annotation as well?

Reply:
doesn't exit in 1.8

Reply:
The annotations and the fields are both constructed from the name and serviceName as below. What information would I want from the existing annotations?